### PR TITLE
[RND-323] Local logging improvements

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: Meadowlark-js/packages/meadowlark-core/src/Logger.ts [hunk 1]
```diff
@@ -6,20 +6,44 @@
 import winston from 'winston';
 import { FrontendRequest } from './handler/FrontendRequest';
 
-const offline = process.env.IS_LOCAL === 'true';
```
Comment (stephenfuqua) [inline Meadowlark-js/packages/meadowlark-core/src/Logger.ts:9]: This was the worst culprit - was always loading before `dotenv` could run.

File: Meadowlark-js/packages/meadowlark-core/src/Logger.ts [hunk 2]
```diff
@@ -6,20 +6,44 @@
...
+
+    if (typeof message === 'object') {
+      // TypeScript thinks that this is a string, but it there are cases where it ends up being an object
+      m = convertErrorToString(m);
+    }
+    if (typeof extra === 'object') {
+      e = JSON.stringify(extra);
+    }
+
+    return `${timestamp} ${level} ${m} ${e}`;
```
Comment (stephenfuqua) [inline Meadowlark-js/packages/meadowlark-core/src/Logger.ts:45]: Now that I've fixed `IS_LOCAL`, the `winston.format.cli()` formatter will ignore the `extra` stuff. This configuration has output like this:

```
2022-07-26 17:29:76 info Server listening at http://127.0.0.1:3000
2022-07-26 17:29:77 info Starting Meadowlark API at http://127.0.0.1:3000/local
2022-07-26 17:29:67 info Request {"url":"/local/v3.3b/ed-fi/contentClassDescriptors}
```

File: Meadowlark-js/packages/meadowlark-core/src/Logger.ts [hunk 3]
```diff
@@ -57,17 +82,15 @@
...
+  // This object is tuned for use in many situations without further customization. For example, it can be used directly in
+  // Fastify, so long as it continues to have definitions for: fatal, error, warn, info, debug, trace, child.
 
-    logger.error({ message, error, traceId });
+  fatal: (message: string, err?: any | null) => {
+    logger.error({ message, err: convertErrorToString(err) });
+    process.exit(1);
+  },
+  error: (message: string, traceId: string | null, err?: any | null) => {
+    logger.error({ message, error: convertErrorToString(err), traceId });
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply Meadowlark-js/packages/meadowlark-core/src/Logger.ts:93]: I need to study this more to see what's going on.

[the remark this answers is not in the document]
Comment (stephenfuqua) [reply Meadowlark-js/packages/meadowlark-core/src/Logger.ts:93]: We already added a function to mask the password at the time of logging. I am going to tweak that to better avoid this. Don't want to create a situation where we have a _different_ unmasked password come through, think it is this one, and then dismiss the alert prematurely.

[the remark this answers is not in the document]
Comment (stephenfuqua) [reply Meadowlark-js/packages/meadowlark-core/src/Logger.ts:93]: Dismissing this alert; it doesn't seem to be accounting for my latest update above.

**Review by bradbanister — APPROVED** [review]
Nodded my head and said "yep" out loud a lot.

**Pull request merged**